crossbeam-skiplist/src/base.rs

```diff
@@ -488,22 +488,28 @@ where
     }
 
     /// Returns an iterator over a subset of entries in the skip list.
-    pub fn ref_range<'a, 'k, Min, Max>(
+    pub fn ref_range<'a, 'g, T, R>(
         &'a self,
-        lower_bound: Bound<&'k Min>,
-        upper_bound: Bound<&'k Max>,
-    ) -> RefRange<'a, 'k, Min, Max, K, V>
+        range: R,
+        guard: &'g Guard,
+    ) -> RefRange<'a, K, V>
     where
-        K: Ord + Borrow<Min> + Borrow<Max>,
-        Min: Ord + ?Sized + 'k,
-        Max: Ord + ?Sized + 'k,
+        K: Borrow<T>,
+        R: RangeBounds<T>,
+        T: Ord + ?Sized,
     {
+        let mut range = self.range(range.start_bound(), range.end_bound(), guard);
+        let head = try_pin_loop(|| range.next());
+        let tail = try_pin_loop(|| range.next_back());
```

Comment:
I'm not too happy about this one: `next_back` is slow to calculate and was not needed in the previous iterator implementation.

Reply:
Unfortunately, I don't know how to remove that `next_back()`. :(

I was hoping that we'd be able to simplify the type soup in `RefRange<'a, Min, Max, K, V>`, but I don't really see a way how to without paying some perf cost. Perhaps those types are not an issue because in practice no one ever spells out these iterator types anyway (especially now that we have `impl Trait`)?

If you don't see a solution here either, I'll put `Min` and `Max` types back into `RefRange`.